src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -11160,20 +11154,7 @@ void Dummy()
 ";
             var compilation = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
 
-            compilation.VerifyDiagnostics(
-                 // (19,49): warning CS8608: Nullability of reference types in type doesn't match overridden member.
```

Comment:
>// (19,49): warning CS8608: Nullability of reference types in type doesn't match overridden member. [](start = 17, length = 99)

I would expect this warning to remain. Consumers accessing A.E2 can pass nullable values to add/remove, B1.E2 accepts only non-nullable values.